sources/views/contacts.js

```diff
@@ -16,34 +17,64 @@ export default class Contact extends JetView {
 				},
 
 				deleteIcon: "<span class='remove-btn webix_icon wxi-close'></span>"
+			},
+			onClick: {
+				"wxi-close": (e, id) => this.deleteContact(id)
 			}
 		};
 
-		const userForm = {
-			view: "form",
-			localId: "userForm",
-			elements: [
-				{
-					view: "text",
-					label: "User Name:",
-					name: "name"
-				},
-				{
-					view: "text",
-					label: "Email:",
-					name: "Email"
-				},
-				{}
-			]
+		const addButton = {
+			view: "button",
+			value: "Add",
+			click: () => this.addContact()
 		};
 
 		const view = {
 			margin: 20,
-			cols: [usersList, userForm]
+			cols: [{ rows: [usersList, addButton] }, userForm]
 		};
 		return view;
 	}
-	init() {
-		this.$$("userList").parse(contacts);
+	init(view, url) {
+		this.list = this.$$("userList");
+		this.list.sync(contacts);
+		this.list.attachEvent("onAfterSelect", chosenId => {
+			const previousId = this.getParam("id");
+			const item = contacts.getItem(previousId);
+
+			if (item) {
+				if (!this.formValidation(item)) {
+					this.deleteContact(previousId);
+				}
+			}
+			this.show(`./contacts?id=${chosenId}`);
+		});
+		let id = url[0].params.id;
+		if (!contacts.exists(id)) {
+			id = contacts.getFirstId();
+		}
+		this.list.select(id);
+	}
+
+	addContact() {
+		const id = contacts.add({ Name: "", Email: "" }, 0);
+		webix.message({ type: "info", text: "Please enter user details" });
+		this.list.select(id);
+	}
+
+	deleteContact(id) {
+		contacts.remove(id);
+		this.list.unselectAll();
+		this.show("./contacts");
+	}
+
+	formValidation(obj) {
```

Comment:
what good does this method do? `obj[key] !== "` means that an empty item with some id will still be valid, so it will never be deleted (and see comment above - we must not delete anything in the `onAfterSelect`)

Reply:
Ready to review.